lib/Doctrine/DBAL/Types/JsonType.php

```diff
@@ -43,6 +43,10 @@ public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $pla
     public function convertToDatabaseValue($value, AbstractPlatform $platform)
     {
         if (null === $value) {
+            if ($platform->getName() == 'oracle') {
+                return '';
+            }
+
```

Comment:
@soyuka the issue here that triggers `ORA-24816` is not that we cannot bind `null` to a `CLOB` type column (which works) but in fact this appears if do an insert like this:

```
Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'INSERT INTO DBAL2386 VALUES (:id, :title, :address, :geo, :name)' with params [1, null, null, null, null]:

ORA-24816: Expanded non LONG bind data supplied after actual LONG or LOB column
```

What works would be the following:

```
INSERT INTO DBAL2386 VALUES (:id, :title, :address, :geo, :name)' with params [1, <anything_including_null>, <anything_including_null>, <anything_including_null>, "test"]
```

So what PHP7 oci8 does not "like" (or I would even say it's a bug because it worked in PHP5) is having a `null` value bound to a column that comes after a LONG/LOB column (the order of the insert list is important here, not the order of binding).
I am not sure what we could possibly do about this.